lib/patch-series.ts

```diff
@@ -231,31 +262,69 @@ export class PatchSeries {
         return mbox.split(separatorRegex);
     }
 
-    protected static removeDuplicateMimeVersionHeaders(mails: string[]): void {
+    protected static removeDuplicateHeaders(mails: string[]): void {
         mails.map((mail: string, i: number) => {
             const endOfHeader = mail.indexOf("\n\n");
             if (endOfHeader < 0) {
                 return;
             }
+
             let headers = mail.substr(0, endOfHeader + 1);
-            const needle = "\nMIME-Version: 1.0\n";
-            const offset = headers.indexOf(needle);
-            if (offset < 0) {
-                return;
-            }
-            let offset2 = headers.indexOf(needle, offset + 1);
-            if (offset2 < 0) {
-                return;
-            }
-            do {
-                headers = headers.substr(0, offset2 + 1)
-                    + headers.substr(offset2 + needle.length);
-                offset2 = headers.indexOf(needle, offset + 1);
-            } while (offset2 > 0);
+            singletonHeaders.forEach((header: ISingletonHeader) => {
+                headers = PatchSeries.stripDuplicateHeaders(headers, header);
+            });
+
             mails[i] = headers + mail.substr(endOfHeader + 1);
         });
     }
 
+    protected static stripDuplicateHeaders(headers: string,
+                                           header: ISingletonHeader): string {
+        const needle = "\n" + header.key + ":";
+        let offset;
+
+        if (headers.startsWith(`${header.key}:`)) {
+            offset = 0;
+        } else {
+            offset = headers.indexOf(needle) + 1;
+            if (!offset) {
+                return headers;
+            }
+        }
+
+        let endofKey = offset + needle.length - 1;
+        offset = headers.indexOf(needle, endofKey + 1);
+
+        if (offset < 0) {
+            return headers;
+        }
```

Comment:
Now we check if `needle` appears a second time, following the first instance. If it doesn't, there's no duplicate, so we're done and exit early.

However, I think this logic (due to the `+1`) can fail in a pathological case, such as input like this:

```
MIME-Version:\n
MIME-Version: 1.0\n
```

In this case, `endofKey` points at the newline following the `:`, and the `+1` bumps it past the newline, so it won't match `needle` even though there clearly is a second instance of the header. The easy fix is simply to drop the `+1`.

Reply:
Removing the +1.